lib/ffmpeg/decoderbase.py

```diff
@@ -43,9 +43,15 @@ def gen_name(self):
       name += "_r2r"
     return name
 
+  def map_best_hw_format(self):
+    return map_best_hw_format(self.format, self.caps["fmts"])
+
+  def mapformat(self):
+    return mapformat(self.format)
+
```

Comment:
I guess this creates the same issue as discussed in #386.  Thus, we better wait for the design fix in #388 to ensure we can override it properly here too.